### PR TITLE
Added support for fenced code blocks.

### PR DESCRIPTION
**Pull request by Witiko**
This commit adds support for fenced code blocks as discussed in #18.

### CODE REVIEW

**Comment by Witiko** [comment]
On a related note, I also added Pandoc-style citation support to [witiko/markdown](/witiko/markdown). The implementation, however, does not keep track of available citations and just transform any citation tokens. This works well for TeX formats, but other Lunamark writers will require an equivalent of the Pandoc's `citeproc` module to produce sane output.

If you were interested in the extension, I can backport it as-is with the extension disabled in writers without the corresponding writer method. Alternatively, I am willing to participate in the implementation of full-fledged citation support, if such support is planned for Lunamark.

File: README.markdown [hunk 1]
```diff
@@ -39,6 +39,7 @@ Lunamark's markdown parser currently supports a number of extensions
   - Footnotes
   - Definition lists
   - Pandoc-style title blocks
+  - Fenced code lists
```
Comment (jgm) [inline README.markdown]: You mean "Fenced code blocks" I assume?

**Comment by jgm** [comment]
Citation support would be nice.  For now it could just pass through literally for non-LaTeX formats, I suppose.

File: README.markdown (hunk 1 above)
Comment (Witiko) [inline README.markdown]: Indeed. 7e10c75 fixes this.

**Comment by Witiko** [comment]
Another thing that bears consideration is how to implement switching between core LaTeX / natbib / BibLaTeX markup. I am thinking of introducing boolean / string options. If the user specifies `options.citations = true`, then they let the writer decide which markup should be used. If the user specifies `options.citations = "..."`, then they assume direct control over the produced markup.

**Comment by Witiko** [comment]
Nobody seems to have noticed/complained yet, but [a subtle bug](https://github.com/Witiko/markdown/issues/14) has been introduced with the added support for fenced code blocks. When a fenced code block appears inside a list and the code begins with an asterisk/hyphen, then the `ListBlockLine` parser will fail to match it and it will be parsed as a nested list. I will do my best to fix the problem and send a patch later this week, but it might happen that I do not get to it until the next week.

**Comment by Witiko** [comment]
I apologize for the false alarm. It seems that there is no consensus on how this case should be handled, but the current lunamark implementation yields the same result as Pandoc.

**Pull request merged**